### PR TITLE
Add 'wth' claim to WPT

### PR DESCRIPTION
**Pull request by arndt-s**
Adds 'wth' claim to WPT to bind WPT to WITs.

### CODE REVIEW

File: draft-ietf-wimse-s2s-protocol.md [hunk 1]
```diff
@@ -313,6 +313,8 @@ A WPT contains the following:
     * `exp`: The expiration time of the WIT (as defined in {{Section 4.1.4 of RFC7519}}). WPT lifetimes MUST be short,
      e.g., on the order of minutes or seconds.
     * `jti`: A unique identifier for the token.
+    * `wth`: Hash of the Workload Identity Token. The value, as defined in {{to-wit}}, is the base64url encoding of the SHA-256
+     hash of the ASCII encoding of the token's value.
```
Comment (yaronf) [inline draft-ietf-wimse-s2s-protocol.md:317]:
```suggestion
    * `wth`: Hash of the Workload Identity Token, defined in {{to-wit}}. The value is the base64url encoding of the SHA-256
     hash of the ASCII encoding of the token's value.
```

The token is defined in {{to-wit}}, not the hash.

**Review by yaronf — CHANGES_REQUESTED** [review]
Editorial

File: draft-ietf-wimse-s2s-protocol.md (hunk 1 above)
Comment (jsalowey) [reply draft-ietf-wimse-s2s-protocol.md:317]: Yaron's suggestion looks good to me.

**Comment by yaronf** [comment]
@arndt-s please also update the relevant IANA section so we don't forget.

**Review by yaronf — APPROVED** [review]

File: draft-ietf-wimse-s2s-protocol.md [hunk 2]
```diff
@@ -351,11 +354,12 @@ The decoded JWT claims of the WPT from the example above are shown here:
...
- "iss": "wimse://example.com/specific-workload",
- "aud": "https://service.example.com/path",
- "exp": 1717612820,
- "jti": "__bwc4ESC3acc2LTC1-_x",
- "ath": "CL4wjfpRmNf-bdYIbYLnV9d5rMARGwKYE10wUwzC0jI"
+  "ath": "CL4wjfpRmNf-bdYIbYLnV9d5rMARGwKYE10wUwzC0jI",
+  "aud": "https://service.example.com/path",
+  "exp": 1728658672,
+  "iss": "wimse://example.com/specific-workload",
+  "jti": "4b42c5f611e2b1cfa1d2c41b3a2fb782",
```
Comment (bc-pi) [inline draft-ietf-wimse-s2s-protocol.md:361]: Whhhhhhyyyyyyyy?  Hex in jti values is so aesthetically displeasing.

**Comment by bc-pi** [comment]
I realize I missed a meeting but why did we add this?

**Comment by bc-pi** [comment]
What's the threat? Different WITs with the same cnf key and a compromised middle thing?

**Comment by bc-pi** [comment]
I just don't see it...

**Comment by bc-pi** [comment]
Iss in the WPT is even less useful with this too.

File: draft-ietf-wimse-s2s-protocol.md (hunk 2 above)
Comment (arndt-s) [reply draft-ietf-wimse-s2s-protocol.md:361]: If you feel strongly about this I can provide a new PR that changes this identifier ;)

**Comment by arndt-s** [comment]
> What's the threat? Different WITs with the same cnf key and a compromised middle thing?

Yes, currently we don't have strong binding between WPT and WIT. Only the fact that `WIT sub == WIT iss`.

> Iss in the WPT is even less useful with this too.

I don't understand the "even less" part but I see that this is somewhat duplicated now.

**Comment by yaronf** [comment]
> Yes, currently we don't have strong binding between WPT and WIT. Only the fact that WIT sub == WIT iss.

Should be: WIT sub == WPT iss.

And just to state it explicitly: people are likely to extend the WIT with various "junk" (i.e., policy) and that wouldn't be covered otherwise.

File: draft-ietf-wimse-s2s-protocol.md (hunk 2 above)
Comment (bc-pi) [reply draft-ietf-wimse-s2s-protocol.md:361]: I do kinda. `__bwc4ESC3acc2LTC1-_x` was okay. and the order of the claims was more readable before

**Pull request merged**